test/resources/test_survey.py

```diff
@@ -165,3 +171,43 @@ def test_get_survey_by_short_name_ce_empty(self, mock_request):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response_data['collection_exercises'], [])
         self.assertEqual(response_data['survey']['id'], "cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87")
+
+    @requests_mock.mock()
+    def test_edit_survey_details_success(self, mock_request):
+        mock_request.put(url_update_survey_details, status_code=200)
+        url = f'backstage-api/v1/survey/edit-survey-details/{survey_ref}'
+        response = self.app.put(url, headers=self.headers, data=json.dumps({
+            "long_name": 'Survey Test Name',
+            "short_name": 'QBZ'
+        }))
+        self.assertEqual(response.status_code, 200)
+
+    @requests_mock.mock()
+    def test_edit_survey_details_failure(self, mock_request):
+        mock_request.put(url_update_survey_details, status_code=500)
+        url = f'backstage-api/v1/survey/edit-survey-details/{survey_ref}'
+        response = self.app.put(url, headers=self.headers, data=json.dumps({
+            "long_name": 'Survey Test Name',
+            "short_name": 'QBZ'
+        }))
+        self.assertEqual(response.status_code, 500)
+
+    @requests_mock.mock()
+    def test_edit_survey_details_not_found(self, mock_request):
+        mock_request.put(url_update_survey_details, status_code=404)
+        url = f'backstage-api/v1/survey/edit-survey-details/{survey_ref}'
+        response = self.app.put(url, headers=self.headers, data=json.dumps({
+            "long_name": 'Survey Test Name',
+            "short_name": 'QBZ'
+        }))
+        self.assertEqual(response.status_code, 404)
+
+    @requests_mock.mock()
+    def test_edit_survey_details_no_data(self, mock_request):
```

Comment:
Are we trying to test the restplus validation here? if so we don't need the mock and the response status we'd want to check for is a 400